# a bunch of things to do...
**Basic Linux questions**
- how do I get colors set up correct for bash shell?
- how do I properly edit/add to PATH? just in .bashrc?
- what are the differences between .bashrc, .profile, and .bash_profile?
- how do I connect Jupyter notebook to Pinky?
- how do I connect R studio to Pinky?


**Guano work**
- demultiplex IM4 from lib05 to determine any potential contaminants  
- if lucky, look at Lib07 (if Steve gets it done) to determine how to proceed with mock communities, water, primers, for Lib08… IF NOT… then just make another batch of mock IM4… add to 2 sheets and use R script to generate list of sampleIDs/barcodeIDs for Steve and submit.  
- follow up with Jeff/Steve to figure out when Lib07 is running, and if PE150 run is happening

**Bat exomes**
-recalculate theta with different parameters... can we reduce .bam files to only targets ID'd from .vcf files? ... maybe a bedtools application? ... want ShinyApp plots with data that is consistent with filtering...
-cut out data from /leo for this project as needed
-work on scripting in Python if possible
-identify mistakes in fastStructure if possible
-MYLU historic samples - want an ancestral genome... can we pool all the Myotis in an area and determine a best representative genome for a given area by pooling a population and picking a base pair? angsd does this, I think in the genotype calling phase
- need to try to get fastMe working again; Toni W. put it on Pinky
-run PCA on angsd data to suss out population structure?


**Writing**
- continue with guide for lib prep and target capture in Github 
